src/transaction/Transaction.js

```diff
@@ -851,6 +851,86 @@ export default class Transaction extends Executable {
         return this;
     }
 
+    /**
+     * This method removes a signature from the transaction based on the public key or signature provided.
+     *
+     * @param {PublicKey} publicKey - The public key associated with the signature to remove.
+     * @param {Uint8Array} [signature] - The signature to remove. If omitted, all signatures for the given public key will be removed.
+     * @returns {Uint8Array[]} The removed signatures.
+     */
+    removeSignature(publicKey, signature) {
+        // If the transaction isn't frozen, freeze it.
+        if (!this.isFrozen()) {
+            this.freeze();
+        }
+
+        const publicKeyData = publicKey.toBytesRaw();
+        const publicKeyHex = hex.encode(publicKeyData);
+
+        if (!this._signerPublicKeys.has(publicKeyHex)) {
+            throw new Error("The public key has not signed this transaction");
+        }
+
+        /** @type {Uint8Array[]} */
+        const removedSignatures = [];
+
+        // Iterate over the signed transactions and remove matching signatures
+        for (const transaction of this._signedTransactions.list) {
```

Comment:
Is the order of the _signedTransaction list locked at this point and unchangeable? If the order is trusted to remain the same throughout the entire add/remove signature calls, then the Uint8Array[] is a good structure for the return value. If the list order isn't trustable, then this structure could be problematic. 
Does the list remain locked even if the transaction is converted to bytes and back into a transaction? Or does the lock get lost in the conversion.

Reply:
The concern about the order of the _signedTransaction list in the context of serialization and deserialization is valid. Based on Hedera's transaction model, when a transaction is serialized into bytes (using .toBytes()) and then deserialized back (using Transaction.fromBytes()), the transaction structure—including the order of signatures—should be preserved.

However, once a transaction is frozen using freezeWith() or freeze(), it becomes immutable. This immutability should include locking the order of signatures within the transaction, ensuring that the order remains consistent through the entire lifecycle of the transaction, including during serialization and deserialization.

I will make further investigations to be 100% sure on this.

Reply:
My only concern is the transfer transaction. When you convert it to bytes it sorts the transactions inside and it might cause some issues. If you can please check if transfer transaction with different token ids works as expected. Keep in mind that the order of `.addTokenTransfer` should be not... well, in order :D So if there is a token with id `0.0.1`, `0.0.2` and `0.0.3` don't pass the token transfers:
```javascript
addTokenTransfer(0.0.1, sender, receiver, amount)
addTokenTransfer(0.0.2 sender, receiver, amount)
addTokenTransfer(0.0.3, sender, receiver, amount)
```